File: ExampleObjc/DemoApp/ViewController.m
//
//  ViewController.m
//  DemoApp
//
//  Created by Deniss Kaibagarovs on 24/10/2022.
//

#import "ViewController.h"
#import <dojo_ios_sdk_drop_in_ui-Swift.h>

@interface ViewController ()
@property DojoSDKDropInUI *dojoUI;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}

- (IBAction)onStartPaymentFlowPress:(id)sender {
    self.dojoUI = [[DojoSDKDropInUI alloc] init];
    DojoThemeSettings *theme = [DojoThemeSettings getLightTheme];
    DojoUIApplePayConfig *appleConfig = [[DojoUIApplePayConfig alloc] initWithMerchantIdentifier: @"merchant-identifier"];
    [self.dojoUI startPaymentFlowWithPaymentIntentId: @"payment-intent-id"
                                          controller: self
                                      customerSecret: nil
                                      applePayConfig: appleConfig
                                       themeSettings: theme
                                          completion: ^(NSInteger result) {
        NSLog(@"%ld", (long)result);
    }];
}

@end

File: ExampleSwift/Example/ViewController.swift
//
//  ViewController.swift
//  dojo-ios-sdk-drop-in-ui
//
//  Created by Den on 08/01/2022.
//  Copyright (c) 2022 Den. All rights reserved.
//

import UIKit
import dojo_ios_sdk_drop_in_ui
import dojo_ios_sdk

class ViewController: UIViewController {
    
    let dojoUI = DojoSDKDropInUI()
    
    @IBAction func onStartPaymentFlowPress(_ sender: Any) {
        let paymentIntentId = "pi_sandbox_mDvzElFkoU2QH440cwoOEg"
        let customerSecret = ""
        let applePayConfig = DojoUIApplePayConfig(merchantIdentifier: "merchant.uk.co.paymentsense.sdk.demo.app")
        dojoUI.startPaymentFlow(paymentIntentId: paymentIntentId,
                                controller: self,
                                customerSecret: customerSecret,
                                applePayConfig: applePayConfig) { result in
            print("SDK result code: \(result)")
        }
    }
}

